x/protorev/keeper/rebalance.go

```diff
@@ -9,40 +9,48 @@ import (
 
 // IterateRoutes checks the profitability of every single route that is passed in
 // and returns the optimal route if there is one
-func (k Keeper) IterateRoutes(ctx sdk.Context, routes []poolmanagertypes.SwapAmountInRoutes) (sdk.Coin, sdk.Int, poolmanagertypes.SwapAmountInRoutes) {
+func (k Keeper) IterateRoutes(ctx sdk.Context, routes []RouteMetaData) (sdk.Coin, sdk.Int, poolmanagertypes.SwapAmountInRoutes) {
 	var optimalRoute poolmanagertypes.SwapAmountInRoutes
 	var maxProfitInputCoin sdk.Coin
 	maxProfit := sdk.ZeroInt()
 
-	for _, route := range routes {
-		// Find the max profit for the route using the token out denom of the last pool in the route as the input token denom
-		inputDenom := route[len(route)-1].TokenOutDenom
-		inputCoin, profit, err := k.FindMaxProfitForRoute(ctx, route, inputDenom)
-		if err != nil {
-			k.Logger(ctx).Error("Error finding max profit for route: ", err)
+	// Get the total number of pool points that can be consumed in this transaction
```

Comment:
Nice job on all the code comments in this file, quite easy to follow the logic :)